File: GridView/GridViewEditDialog/GridViewEditDialogCSharp/EditForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Telerik.WinControls;

namespace GridViewEditDialog
{
    public partial class EditForm : Telerik.WinControls.UI.RadForm
    {
        private MainForms.Student student;

        public EditForm()
        {
            InitializeComponent();
        }

        public EditForm(MainForms.Student student) : this()
        {
            this.student = student;

            this.radDropDownList1.Items.AddRange(new List<string>() { "A", "A+", "A-", "B", "B+", "B-", "C", "C+", "C-", "D", "D+", "D-", "E", "E+", "E-", "F", "F+", "F-" });
            this.radDropDownList1.DropDownStyle = RadDropDownStyle.DropDownList;
            this.radDropDownList1.DropDownListElement.SyncSelectionWithText = true;

            this.radSpinEditor1.Value = this.student.Id;
            this.radTextBox1.Text = this.student.Name;
            this.radDropDownList1.SelectedValue = this.student.Grade;
        }

        private void save_Button_Click(object sender, EventArgs e)
        {
            if (IsValidData())
            {
                this.student.Id = (int)this.radSpinEditor1.Value;
                this.student.Name = this.radTextBox1.Text;
                this.student.Grade = this.radDropDownList1.SelectedItem.Text;
                this.Close();
            }
        }

        private bool IsValidData()
        {
            this.errorProvider1.Clear();

            if (this.radTextBox1.Text == string.Empty)
            {
                this.errorProvider1.SetError(this.radTextBox1, "Invalid name!");
                return false;
            }

            return true;
        }

        private void cancel_Button_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}

File: GridView/GridViewEditDialog/GridViewEditDialogVB/EditForm.vb
Imports System.ComponentModel
Imports System.Text
Imports Telerik.WinControls

Namespace GridViewEditDialog
	Partial Public Class EditForm
		Inherits Telerik.WinControls.UI.RadForm
		Private student As MainForms.Student

		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub New(ByVal student As MainForms.Student)
			Me.New()
			Me.student = student

			Me.radDropDownList1.Items.AddRange(New List(Of String)() From {"A", "A+", "A-", "B", "B+", "B-", "C", "C+", "C-", "D", "D+", "D-", "E", "E+", "E-", "F", "F+", "F-"})
			Me.radDropDownList1.DropDownStyle = RadDropDownStyle.DropDownList
			Me.radDropDownList1.DropDownListElement.SyncSelectionWithText = True

			Me.radSpinEditor1.Value = Me.student.Id
			Me.radTextBox1.Text = Me.student.Name
			Me.radDropDownList1.SelectedValue = Me.student.Grade
		End Sub

		Private Sub save_Button_Click(ByVal sender As Object, ByVal e As EventArgs) Handles save_Button.Click
			If IsValidData() Then
				Me.student.Id = CInt(Fix(Me.radSpinEditor1.Value))
				Me.student.Name = Me.radTextBox1.Text
				Me.student.Grade = Me.radDropDownList1.SelectedItem.Text
				Me.Close()
			End If
		End Sub

		Private Function IsValidData() As Boolean
			Me.errorProvider1.Clear()

			If Me.radTextBox1.Text = String.Empty Then
				Me.errorProvider1.SetError(Me.radTextBox1, "Invalid name!")
				Return False
			End If

			Return True
		End Function

		Private Sub cancel_Button_Click(ByVal sender As Object, ByVal e As EventArgs) Handles cancel_Button.Click
			Me.Close()
		End Sub
	End Class
End Namespace